### PR TITLE
State guard aop

### PR DESCRIPTION
**Pull request by jdillon**

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/stateguard/StateGuardModule.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.google.inject.AbstractModule;
+import com.google.inject.matcher.Matchers;
+
+/**
+ * State guard module.
+ *
+ * @since 3.0
+ */
+public class StateGuardModule
+  extends AbstractModule
```
Comment (mcculls) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/stateguard/StateGuardModule.java]: Extend AbstractInterceptorModule and add @Named if you want this automatically installed across all Nexus bundles.

**Comment by mcculls** [comment]
+1

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/stateguard/StateGuardModule.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/stateguard/StateGuardModule.java]: Hrm... can't as AbstractInterceptorModule is in nexus-core and nexus-repository does not depend on it, it depends on nexus-repository.

Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/stateguard/StateGuardModule.java]: Okay I may copy this over to goodies-inject... though goodies needs some general maintainance and upgrading to java7 which I'm not looking forward too... but neds to be done one day.

**Comment by mcculls** [comment]
looks good to go

**Pull request merged**